python/Ganga/GPIDev/Adapters/IBackend.py

```diff
@@ -511,3 +513,32 @@ def updateMonitoringInformation(jobs):
         """
 
         raise NotImplementedError
+
+
+def group_jobs_by_backend_credential(jobs):
+    # type: (List[Job]) -> List[List[Job]]
+    """
+    Split a list of jobs based on the credential required by their backends.
+
+    Any missing or invalid credentials are added to ``needed_credentials``
+
+    Args:
+        jobs: a list of ``Job``s
+
+    Returns:
+        a list of list of ``Job``s with each sublist sharing a credential
+    """
+    jobs_by_credential = defaultdict(list)
+    for j in jobs:
+        try:
+            cred_req = j.backend.credential_requirements
+            cred = credential_store[cred_req]
+            if not cred.is_valid():
+                logger.debug('Required credential %s is not valid', cred)
+                needed_credentials.add(cred_req)
+                continue
+            jobs_by_credential[cred].append(j)
+        except KeyError:
```

Comment:
Doing this catches problems from `credential_store` as well as `jobs_by_credential` from reading the code.
Which of these are expected to be checked here?